pkg/webhooks/machine_webhook.go

```diff
@@ -1332,6 +1335,58 @@ func validateMachineLifecycleHooks(m, oldM *machinev1.Machine) []error {
 	return errs
 }
 
+func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProviderSpec, parentPath *field.Path) []error {
+
+	var errs []error
+	dataDiskLuns := make(map[int32]struct{})
+	dataDiskNames := make(map[string]struct{})
+	reg := regexp.MustCompile(`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`)
```

Comment:
Please add a comment to explain what the regex is trying to achieve, reading regexs and interpreting them takes a lot longer than reading a well thought out explanation of the regex

Reply:
Yes I have a comment explaining what it does on the api spec, but forgot to add it here. Thanks for flagging.
Added.